File: 백준/Bronze/14954. Happy Number/Happy Number.cc
#include <cmath>
#include <iostream>
using namespace std;
int hpy(int n) {
    int s = 0;
    while (n > 0) {
        s += pow(n % 10, 2);
        n /= 10;
    }
    return s;
}
int main(void) {
    int n;
    cin >> n;
    while (1) {
        n = hpy(n);
        if (n == 4) {
            cout << "UNHAPPY";
            break;
        } else if (n == 1) {
            cout << "HAPPY";
            break;
        }
    }
    return 0;
}

File: 백준/Bronze/14954. Happy Number/Happy Number.py
import sys
input=lambda:sys.stdin.readline().rstrip()
def hpy(n):
    s=0
    while n>0:
        s+=(n%10)**2
        n//=10
    return s
n=int(input())
while 1:
    n=hpy(n)
    if n==4:
        print("UNHAPPY")
        break
    elif n==1:
        print("HAPPY")
        break